File: python/36c_1520B_OrdinaryNumbers.py
from math import log10 as log

def main():
    ans = []
    t = int(input())
    for _ in range(t):
        n = int(input())
        k = int(log(n) + 1)
        cnt = 0 + (9 * (k -1))
        next_numb = ""
        for i in range(k):
            next_numb += "1"
        next_numb = int(next_numb)
        for j in range(next_numb, n + 1, next_numb):
            cnt += 1
        ans.append(cnt)
    for i in ans:
        print(i)
    
if  __name__ == "__main__":
    main()

File: cpp/36c_1520B_OrdinaryNumbers.cpp
#include<bits/stdc++.h>

int get_next(int k){
    int num = 1;
    for(int i = 1; i < k; i++){
        num = (num * 10) + 1;
    }
    return num;
}

int main(){
    int t, n, k, cnt, numb;
    std::vector<int> ans;
    std::cin>>t;
    for(int i = 0; i < t; i++){
        std::cin>>n;
        k = log10(n) + 1;
        cnt = 0 + (9 * (k - 1));
        numb = get_next(k);
        for(int j = numb; j <= n; j = j + numb){
            cnt++;
        }
        ans.push_back(cnt);
    }
    for(auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}